zipline/pipeline/loaders/events.py

```diff
@@ -148,7 +152,8 @@ def __init__(self,
                 raise ValueError(
                     WRONG_MANY_COL_DATA_FORMAT_ERROR.format(sid=k)
                 )
-
+        self.events_by_sid = {sid: df.dropna(subset=[self.event_date_col]) for
```

Comment:
We conferred last week and I tried the solution he suggested, but something was still broken and required more debugging. Talked to @joshpayne at standup on Friday and he suggested just doing this instead of debugging blaze to fix this (for now).